src/facet_display/FacetDisplay.test.tsx

```diff
@@ -1,118 +1,200 @@
 import React from "react"
-import { shallow } from "enzyme"
+import { render, screen, within } from "@testing-library/react"
+import user from "@testing-library/user-event"
```

Comment:
The vast majority of the line changes in this PR are from removing enzyme in this test file, and adding `@testing-library/react`, `@testing-library/user-event`.

I was going to add more tests in this file, and I did not want to add enzyme tests since they are fragile / don't work with React 17, 18. (In previous PR, I had to alter the test file because I extracted `AvailableFacets` from `FacetDisplay` https://github.com/mitodl/course-search-utils/pull/78/files#diff-0b2ec90271aa9923bd1a4e5747946849aa084238d325cc529cda4848f503b8a3). So I rewrote the enzyme tests with `@testing-library`.

I also added some tests... E.g., I didn't see tests for clicking the expand/collapse buttons.